torch/csrc/lazy/core/shape_inference.cpp

```diff
@@ -454,6 +455,25 @@ std::vector<Shape> compute_shape_slogdet(const at::Tensor & self) {
           Shape(self.scalar_type(), out_sizes)};
 }
 
+std::vector<torch::lazy::Shape> compute_shape_logical_and(at::Tensor & self, const at::Tensor & other) {
+  TORCH_INTERNAL_ASSERT(at::are_expandable(self.sizes(), other.sizes()));
+  return {Shape(c10::ScalarType::Bool, at::infer_size(self.sizes(), other.sizes()))};
+}
+
+std::vector<torch::lazy::Shape> compute_shape_logical_not(at::Tensor & self) {
+  return {Shape(c10::ScalarType::Bool, self.sizes().vec())};
```

Comment:
what was your process for determining the correct shapes for these ops? (why not infer_size here?)

Reply:
AFAIK the `logical_not` takes in one tensor (as opposed to other ops `logical_` ops the take in two tensors `self` and `other`) so `infer_size` is not required here -- the return value is just the element wise `logical not` operation applied to the input tensor.

Reply:
yea that is pretty obvious now that i look at it again :) thanks!